File: backend/src/main/java/entity/Lobby.java
package entity;

import jakarta.persistence.*;
import jakarta.websocket.Session;
import lombok.*;

import jakarta.enterprise.context.ApplicationScoped;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.concurrent.ConcurrentHashMap;

@Getter
@Setter
@Builder
@ApplicationScoped
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "lobbies")
public class Lobby {
    @Id
    @Column(name = "lobby_id")
    private String lobbyId;

    @Transient
    private Map<String, Session> sessions = new ConcurrentHashMap<>();

    @ElementCollection
    @CollectionTable(name = "lobby_users", joinColumns = @JoinColumn(name = "lobby_id"))
    @Column(name = "user_name")
    private Set<String> users = new HashSet<>();


    public Lobby(String lobbyId) {
        this.lobbyId = lobbyId;
    }

    public void AddSession(String userName, Session session){
        sessions.put(userName,session);
    }

    public void AddUserName(String userName){
        users.add(userName);
    }

    public void RemoveSession(String userName){
        sessions.remove(userName);
        users.remove(userName);
    }

    public Set<String> getUserNames(){
        return users;
    }
}

File: Backend/Core/Entities/Lobby.cs
using System.Collections.Concurrent;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Base.Core.Entities;

namespace Core.Entities;

public class Lobby : EntityObject
{
    public string LobbyId { get; set; }

    [NotMapped]
    public ConcurrentDictionary<string, System.Net.WebSockets.WebSocket> Sessions { get; set; } = new ConcurrentDictionary<string, System.Net.WebSockets.WebSocket>();

    [Column("user_name")]
    public List<string> Users { get; set; } = new List<string>();

    // Default constructor
    public Lobby() { }

    // Constructor with lobbyId
    public Lobby(string lobbyId)
    {
        LobbyId = lobbyId;
    }

    public void AddSession(string userName, System.Net.WebSockets.WebSocket session)
    {
        Sessions.TryAdd(userName, session);
    }
    

    public void AddUserName(string userName)
    {
        Users.Add(userName);
    }

    public void RemoveSession(string userName)
    {
        Sessions.TryRemove(userName, out _);
        Users.Remove(userName);
    }

    public List<string> GetUserNames()
    {
        return Users;
    }
    
    public ConcurrentDictionary<string, System.Net.WebSockets.WebSocket> GetSessions()
    {
        return Sessions;
    }
}